src/platform/linux/wayland/event_loop.rs

```diff
@@ -193,7 +193,13 @@ impl EventsLoop {
         let static_cb = unsafe { ::std::mem::transmute(Box::new(callback) as Box<FnMut(_)>) };
         let old_cb = unsafe { self.sink.lock().unwrap().set_callback(static_cb) };
 
-        while !self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
+        loop {
+            // If `interrupt` was called, break from the loop after resetting the flag.
+            if self.interrupted.load(::std::sync::atomic::Ordering::Relaxed) {
+                self.interrupted.store(false, ::std::sync::atomic::Ordering::Relaxed);
```

Comment:
Why not simply set this when entering `run_forever`? This is what X11 and macOS do. Although, It appears macOS takes both the approach in this PR and setting it at the top of the function. Seems a bit redundant.